packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte

```diff
@@ -469,6 +479,58 @@
       : value;
   }
 
+  function formatBalance(value: number, maxDecimals: number = 4): string {
+    return Number.isInteger(value)
+      ? value.toString()
+      : value.toFixed(maxDecimals);
+  }
+
+  async function checkBalance() {
+    try {
+      if (!address || !paymentCurrencies[0] || !network) {
+        console.log("Missing required parameters for balance check:", {
+          address,
+          paymentCurrency: paymentCurrencies[0],
+          network,
+        });
+        return;
+      }
+
+      const invoiceNetworkId = Number(getNetworkIdFromNetworkName(network));
+
+      if (account.chainId !== invoiceNetworkId) {
+        hasEnoughBalance = false;
+        console.log("Wrong network - balance check skipped");
+        return;
+      }
```

Comment:
Why skip the balance check if the user is connected to the wrong chain?

Reply:
Well we are not looking at the right balance on the right network, so we return the function, call the switchNetwork function and then checking the balance again on the right network